### PR TITLE
Third party resources frontend

### PR DESCRIPTION
**Pull request by maciaszczykm**
Works on `gulp serve`:

screenshot at lut 21 15-33-09

screenshot at lut 21 15-33-26

~There is an issue on `gulp serve:prod`, because it doesn't resolve third party resources when navigation is rendered and there are no third party resource menu entries.~

### CODE REVIEW

**Comment by maciaszczykm** [comment]
@bryk Could you take a look on issue with `gulp serve:prod` if it doesn't require much time? What build process may change there?

**Comment by bryk** [comment]
Is this reviewable? Or still WIP?

**Comment by maciaszczykm** [comment]
It's already reviewable. Works on `gulp serve:prod`. I'm going to take care of tests now. You can review.

**Comment by bryk** [comment]
@mijajasinski

File: src/app/frontend/thirdpartyresourcelist/cardlist.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    [[Third Party Resources|Label which appears above the list of such objects.]]
+  </kd-resource-card-list-header>
+  <kd-resource-card-list-pagination pagination-id="thirdpartyresources"
+                                    list="$ctrl.thirdPartyResourceList"
+                                    list-resource="$ctrl.thirdPartyResourceListResource">
+  </kd-resource-card-list-pagination>
+
+  <kd-resource-card-header-columns>
+    <kd-resource-card-header-column size="small" grow="4">
+      [[Name|Label 'Name' which appears as a column label in the table of secrets (secret listview).]]
```
Comment (floreks) [inline src/app/frontend/thirdpartyresourcelist/cardlist.html]: s/secret/third party resource

File: src/app/frontend/thirdpartyresourcedetail/info_component.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @final
+ */
+export default class ThirdPartyResourceInfoComponent {
```
Comment (floreks) [inline src/app/frontend/thirdpartyresourcedetail/info_component.js]: This controller is actually not needed if there is no logic.

File: src/app/frontend/thirdpartyresourcelist/card.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      <kd-labels labels="::$ctrl.thirdPartyResource.objectMeta.labels"></kd-labels>
+    </kd-resource-card-column>
+    <kd-resource-card-column>
+      {{::$ctrl.thirdPartyResource.objectMeta.creationTimestamp | relativeTime}}
+      <md-tooltip>
+        {{::$ctrl.getCreatedAtTooltip($ctrl.thirdPartyResource.objectMeta.creationTimestamp)}}
+      </md-tooltip>
+    </kd-resource-card-column>
+  </kd-resource-card-columns>
+</kd-resource-card>
```
Comment (floreks) [inline src/app/frontend/thirdpartyresourcelist/card.html]: New line

**Comment by maciaszczykm** [comment]
Done.

PTAL

**Comment by maciaszczykm** [comment]
@mijajasinski PTAL

**Comment by floreks** [comment]
You can try out example from docs:
```yaml
apiVersion: extensions/v1beta1
kind: ThirdPartyResource
metadata:
  name: cron-tab.stable.example.com
description: "A specification of a Pod to run on a cron style schedule"
versions:
- name: v1
```

**Comment by maciaszczykm** [comment]
@bryk I'm already working on it. It's not so easy due issues with clients. Each api version needs new client and it's not the only one issue with TPRs at the moment.

**Comment by maciaszczykm** [comment]
Merging it. Instances of the TPR on the details page will follow in next pull request. We can do any needed fixes later too.

**Pull request merged**